### PR TITLE
gh-104389: Add 'unused' keyword to Argument Clinic C converters

### LINKED ISSUE #104389: Make it possible to mark Argument Clinic args as unused
For base classes, it is not unusual to create stubs for interface methods, and let those stubs simply raise an exception. It would be nice to be able to mark the arguments with Py\_UNUSED in those cases.

Implementing this is straight-forward in clinic.py. CConverter get's a new keyword argument `unused` which defaults to `False`, and we can write code like this:

```
/*[clinic input]
_io._TextIOBase.read
    cls: defining_class
    size: int(unused=True) = -1
[clinic start generated code]*/
```

### Linked PRs
* gh-104390

### PR DESCRIPTION
**Pull request by erlend-aasland**
Use the unused keyword param in the converter to explicitly
mark an argument as unused:

    /*[clinic input]
    BaseClass.interface
        flag: bool(unused=True)

* Issue: gh-104389

----
:books: Documentation preview :books:: https://cpython-previews--104390.org.readthedocs.build/

### CODE REVIEW

**Comment by erlend-aasland** [comment]
The \_io change is to show the feature; we can revert it before landing.

**Review by corona10 — APPROVED** [review]
lgtm

**Comment by vstinner** [comment]
Do you have at least one use case for this addition?

**Review by vstinner — APPROVED** [review]
LGTM. It would be nice to have directly an user of this feature, but I'm fine with adding support for it, and later have a separated PR using it. I see where you will use it in the _io extension ;-)

**Comment by erlend-aasland** [comment]
> LGTM. It would be nice to have directly an user of this feature, but I'm fine with adding support for it, and later have a separated PR using it. I see where you will use it in the _io extension ;-)

Yes, I'll create PRs for usage quite soon :)

Anyway, I got two approvals here, a couple of thumbs up on the PR, five likes on the Discourse topic. I'll land this soon.

**Comment by erlend-aasland** [comment]
> LGTM. It would be nice to have directly an user of this feature, but I'm fine with adding support for it, and later have a separated PR using it. I see where you will use it in the _io extension ;-)

PR for \_io usage:

- gh-104418

**Pull request merged**
